src/client/clientlauncher.cpp

```diff
@@ -298,8 +304,13 @@ void ClientLauncher::init_args(GameStartData &start_data, const Settings &cmd_ar
 bool ClientLauncher::init_engine()
 {
 	receiver = new MyEventReceiver();
-	m_rendering_engine = new RenderingEngine(receiver);
-	return m_rendering_engine->get_raw_device() != nullptr;
+	try {
+		m_rendering_engine = new RenderingEngine(receiver);
+	} catch (std::exception &e) {
+		errorstream << e.what() << std::endl;
+		delete m_rendering_engine;
```

Comment:
Missing `m_rendering_engine = nullptr`.

Reply:
actually the delete here is redundant because if `new RenderingEngine` does not return then the value won't be set.
https://isocpp.org/wiki/faq/exceptions#ctors-can-throw